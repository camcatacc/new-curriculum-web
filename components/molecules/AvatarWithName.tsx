import { Avatar } from "@material-ui/core";
import React from "react";
import { InlineStyles } from "../../styles/stylesInterface";
import { AvatarName } from "../atoms/AvatarName";

export const AvatarWithName = (props: { name: string; surname: string }) => {
    return (
        <>
            <div
                style={{ paddingTop: "100%" }}
                className="w-full relative mb-4"
            >
                <Avatar
                    /* The style must be inline instead of tailwind because
                    the material ui would override it */
                    style={{
                        width: "100%",
                        height: "100%",
                        position: "absolute",
                        top: 0,
                        left: 0,
                        bottom: 0,
                        right: 0,
                    }}
                    /* className="w-full h-full absolute top-0 left-0 bottom-0 right-0" */
                    src="/me.jpg"
                />
            </div>
            <AvatarName name={props.name} surname={props.surname} />
        </>
    );
};

const styles: InlineStyles = {
    container: {
        display: "flex",
        flex: 1,
        flexDirection: "column",
    },
    avatarContainer: {
        width: "100%",
        paddingTop: "100%",
        position: "relative",
        marginBottom: "1rem",
    },
    avatar: {
        width: "100%",
        height: "100%",
        position: "absolute",
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
    },
};
