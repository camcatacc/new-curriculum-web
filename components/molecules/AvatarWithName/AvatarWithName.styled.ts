// Modules
import styled from "styled-components";

// Components
export const Container = styled.div`
	padding-top: 100% !important;
`;
