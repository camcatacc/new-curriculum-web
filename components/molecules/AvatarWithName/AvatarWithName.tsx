// Modules
import React from "react";

// Elements
import { Avatar } from "@material-ui/core";
import AvatarName from "components/atoms/AvatarName/AvatarName";

// Definitions
import { InlineStyles } from "styles/stylesInterface";

// Styles
const styles: InlineStyles = {
	container: {
		display: "flex",
		flex: 1,
		flexDirection: "column"
	},
	avatarContainer: {
		width: "100%",
		paddingTop: "100%",
		position: "relative",
		marginBottom: "1rem"
	},
	avatar: {
		width: "100%",
		height: "100%",
		position: "absolute",
		top: 0,
		left: 0,
		bottom: 0,
		right: 0
	}
};

// Definition
export interface AvatarWithNameProps {
	name: string;
	surname: string;
}

// Element
const AvatarWithName = ({ name, surname }: AvatarWithNameProps) => {
	return (
		<>
			<div style={{ paddingTop: "100%" }} className="w-full relative mb-4">
				<Avatar
					/* The style must be inline instead of tailwind because
                    the material ui would override it */
					style={{
						width: "100%",
						height: "100%",
						position: "absolute",
						top: 0,
						left: 0,
						bottom: 0,
						right: 0
					}}
					/* className="w-full h-full absolute top-0 left-0 bottom-0 right-0" */
					src="/me.jpg"
				/>
			</div>
			<AvatarName name={name} surname={surname} />
		</>
	);
};

export default AvatarWithName;
