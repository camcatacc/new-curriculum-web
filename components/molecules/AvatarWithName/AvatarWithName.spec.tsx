// Modules
import React from "react";
import { render } from "@testing-library/react";

// Elements
import AvatarWithName from "components/molecules/AvatarWithName/AvatarWithName";

// Definitions
import type { AvatarWithNameProps } from "components/molecules/AvatarWithName/AvatarWithName";

// Default Props
const defaultProps: AvatarWithNameProps = {
	name: "David",
	surname: "Camacho Cateura"
};

// Tests
describe("AvatarWithName", () => {
	it("Renders as expected", () => {
		const { container } = render(<AvatarWithName {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});
});
