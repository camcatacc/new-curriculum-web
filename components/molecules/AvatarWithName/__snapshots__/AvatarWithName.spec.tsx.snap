// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarWithName Renders as expected 1`] = `
<div>
  <div
    class="sc-gsDKAQ mtNEY w-full relative mb-4"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <noscript />
      <img
        alt="DavidCamacho Cateura"
        class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
        data-nimg="true"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
  </div>
  <h3
    class="MuiTypography-root sc-bdvvtL iytgcZ MuiTypography-h3 MuiTypography-alignCenter"
  >
    David
     
    Camacho Cateura
  </h3>
</div>
`;
